packages/venia-concept/src/components/Header/header.js

```diff
@@ -1,4 +1,6 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
```

Comment:
I'd like @jimbo to chime in here too but from what I've gathered so far we want to be very deliberate where we `connect` components.

In this case, an alternative approach to `connect`ing this component would be to add `itemsCount ` to the connected component `App/container.js`, and pass it down to this component via props (you'll have to pass it through `Main/main.js` too).

There's a balancing act here to be sure but it does simplify things quite nicely when done well.

Reply:
@supernova-at thanks again for all the suggestions :+

1. I get it your point, and indeed it's a balancing act. But isn't that the reason we want to use Redux, i.e. not having to pass props down to children through components that don't require the props at all?  I find passing down props often less clear than just having a connected component in which you just 'select' a state object with mapStateToProps.

2. In a first setup I had this logic in the cartTrigger.js which is already a connected component.  I however moved the cart counter to header.js because I felt that cart counter should not be a part of cartTrigger.js which is an action based component (trigger an action)  This said I even thought about a complete separate cartCounter.js  component which might solve the header component re-render issue that I faced and which is easier to isolate / fix in a single purpose component.

Please let me know what you prefer and I'll refactor. Will refactor anyhow again because I will be adding the items counter back in again, see issue #617

Reply:
@supernova-at see my previous comment ^^, what do you think, shall I add the counter to cartTrigger instead, so we don't introduce an additional `connect`ed component?

Reply:
Sorry for the delay, @ennostuurman - vacation called 😄 

I think absolutely a `cartCounter.js` component would be awesome.

Reply:
@supernova-at I updated PR according to new specs, see https://github.com/magento-research/pwa-studio/issues/617#issuecomment-451650645 . 

Unfortunate a peregrine/package-lock.json made it into this PR, I always try to keep them out, this on e slipped through tho and I couldn't manage to remove it despite rebasing my branch w/ latest release/2.0 .

Could you please review? Thx!

Reply:
@supernova-at FYI => I updated PR with latest staging, fixed header.js conflict (recent react-feather changes) .